evap/results/exporters.py

```diff
@@ -262,12 +291,22 @@ def write_questionnaire(self, questionnaire, evaluations_with_results, contribut
                 approval_count = 0
 
                 for grade_result in results[questionnaire.id]:
-                    if grade_result.question.id != question.id or not grade_result.has_answers:
+                    if grade_result.question.id != question.id or (
+                        isinstance(grade_result, RatingResult) and not grade_result.has_answers
+                    ):
```

Comment:
Again, not happy: This modifies behavior and thus widens the contract of this snippet here. A for "add type annotations" PR, I'd like to have no functional changes.
```suggestion
                    assert isinstance(grade_result, RatingResult)
                    if grade_result.question.id != question.id or not grade_result.has_answers:
```